File: atcoder/abs/5.c
//usr/bin/env picoc $0 - $@;exit
#include <stdio.h>
int main(){
	int n,a,b,r=0,i,j;
	scanf("%d%d%d",&n,&a,&b);
	for(i=1;i<=n;i++){
		int s=0;
		j=i;
		for(;j;j/=10)s+=j%10;
		if(a<=s&&s<=b)r+=i;
	}
	printf("%d\n",r);
	return 0;
}

File: atcoder/abs/5.lua
#!/usr/bin/env lua
local n, a, b = io.read("*n", "*n", "*n")
local r = 0
for i = 1, n do
  local s = 0
  local j = i
  while j > 0 do
    s = s + (j % 10)
    j = j // 10
  end
  if a <= s and s <= b then
    r = r + i
  end
end
return print(r)
